provider/marathon/marathon.go

```diff
@@ -284,7 +312,13 @@ func (p *Provider) getPort(task marathon.Task, application marathon.Application)
 	return strconv.Itoa(port)
 }
 
-func (p *Provider) getWeight(application marathon.Application) string {
+// Extract weight from labels for a given service and a given marathon application
+func (p *Provider) getWeight(application marathon.Application, varg ...string) string {
+	serviceName := extractServiceName(varg...)
+	if value, ok := getApplicationServiceLabel(application, serviceName, "weight"); ok {
+		return value
+	}
+
 	if label, ok := p.getLabel(application, types.LabelWeight); ok {
```

Comment:
since we already have marathon-specific `func (p *Provider) getLabel(application marathon.Application, label string) (string, bool) {` defined, are you ok with extending it's signature to take `serviceName string` as last argument.  So we would check if `serviceName` is not empty, then try to load the service label, otherwise go for application one directly.